driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -221,9 +221,9 @@ public void onResult(final Void result, final Throwable t) {
                             LOGGER.trace(format("Pooled connection %s to server %s is now open",
                                                        pooledConnection.getDescription().getConnectionId(), serverId));
                         }
-                        callback.onResult(pooledConnection, null);
                         connectionPoolListener.connectionCheckedOut(
                                 new ConnectionCheckedOutEvent(pooledConnection.getDescription().getConnectionId()));
+                        callback.onResult(pooledConnection, null);
```

Comment:
What's the effect of this change?

Reply:
See my comment - its really making the ordering the same as the error scenario. Helps ensure the callback side effects don't interact.  Now the closed check in getDescription is removed it doesn't really have an impact.